modules/gapi/samples/one_vpl_source_to_bgr_conv.cpp

```diff
@@ -0,0 +1,203 @@
+#define D3D11_NO_HELPERS
+#define NOMINMAX
+#include <d3d11.h>
+#pragma comment(lib, "dxgi")
+#undef NOMINMAX
+#undef D3D11_NO_HELPERS
+
+#endif // HAVE_D3D11
+#endif // HAVE_DIRECTX
+
+namespace cfg {
+typename cv::gapi::wip::onevpl::CfgParam create_from_string(const std::string &line);
+}
+
+namespace {
+template <typename COMNonManageableType>
+void release(COMNonManageableType *ptr) {
+    if (ptr) {
+        ptr->Release();
+    }
+}
+template <typename COMNonManageableType>
+using ComPtrGuard = std::unique_ptr<COMNonManageableType, decltype(&release<COMNonManageableType>)>;
+
+template <typename COMNonManageableType>
+ComPtrGuard<COMNonManageableType> createCOMPtrGuard(COMNonManageableType *ptr = nullptr) {
+    return ComPtrGuard<COMNonManageableType> {ptr, &release<COMNonManageableType>};
+}
+
+#ifdef HAVE_DIRECTX
+#ifdef HAVE_D3D11
+using AccelParamsType = std::tuple<ComPtrGuard<ID3D11Device>, ComPtrGuard<ID3D11DeviceContext>>;
+
+AccelParamsType create_device_with_ctx(IDXGIAdapter* adapter) {
+    UINT flags = 0;
+    D3D_FEATURE_LEVEL feature_levels[] = { D3D_FEATURE_LEVEL_11_1,
+                                           D3D_FEATURE_LEVEL_11_0,
+                                         };
+    D3D_FEATURE_LEVEL featureLevel;
+    ID3D11Device* ret_device_ptr = nullptr;
+    ID3D11DeviceContext* ret_ctx_ptr = nullptr;
+    HRESULT err = D3D11CreateDevice(adapter, D3D_DRIVER_TYPE_UNKNOWN,
+                                    nullptr, flags,
+                                    feature_levels,
+                                    ARRAYSIZE(feature_levels),
+                                    D3D11_SDK_VERSION, &ret_device_ptr,
+                                    &featureLevel, &ret_ctx_ptr);
+    if (FAILED(err)) {
+        throw std::runtime_error("Cannot create D3D11CreateDevice, error: " +
+                                 std::to_string(HRESULT_CODE(err)));
+    }
+
+    return std::make_tuple(createCOMPtrGuard(ret_device_ptr),
+                           createCOMPtrGuard(ret_ctx_ptr));
+}
+#endif // HAVE_D3D11
+#endif // HAVE_DIRECTX
+}
+
+const std::string about =
+    "This is an example presents decoding on GPU using VPL Source and passing it to OpenCL backend";
+const std::string keys =
+    "{ h help   |                           | Print this help message }"
+    "{ input    |                           | Path to the input video file. Use .avi extension }"
+    "{ output   | output.avi                | Path to the output RAW video file. Use .avi extension }";
+
+int main(int argc, char *argv[]) {
+    cv::CommandLineParser cmd(argc, argv, keys);
+    cmd.about(about);
+    if (cmd.has("help")) {
+        cmd.printMessage();
+        return 0;
+    }
+#ifdef HAVE_DIRECTX
+#ifdef HAVE_D3D11
+    // Get file name
+    const auto input = cmd.get<std::string>("input");
+    const auto output = cmd.get<std::string>("output");
+
+    auto adapter_factory = createCOMPtrGuard<IDXGIFactory>();
```

Comment:
Code from there (`101`) until (`138`) shouldn't be in sample. We must stop using `DIRECTX`/`D3D11` specific code in our samples.

@SergeyIvanov87 Can it be hidden in some `default` GPU implementation for `oneVPL` source? I'm ok if it's not flexible and efficient.

Replace everything with:
```
cap = cv::gapi::make_onevpl_src(input, source_cfgs, "GPU", cv::gapi::onevpl::default_gpu_device());
```

Reply:
At least until we understand how to make it nicer

Reply:
1. These details aren't acceptable for `OMZ` demos at least.
2. Since no default implementation not every user can use at least base functionality. I'd say, `G-API` users would be happy not know such details.
3. It's really `overload` sample readability. Moreover it needs to be copy-pasted to every sample that use onevpl.
4. Not acceptable from python

I don't mind to have this `configurable` way, but let's have both approaches (`default`/`configurable`) and keep this low level code as far as possible from user until it's really necessary...

Reply:
I got your point, but i cannot agree with it:
1) then let's rename it `OMZ_demo`, not just a general `onevpl_source_to_bgr_conv` sample

2) don't say for users. The 'Users' usually want to solve their specific problems or find a way how to do that.
So if samples doesn't answer these questions then it is a bad sample.
The most complicated questions in 3PP ( like openCV/G-API) are integration problem. By integration I mean - adopt this 3PP into own-self-developed-framework. So, the major aspects of integration must be reflected in samples. And device selection is the major aspect here.

3) we already has a lot of code which copied across devices. But I agree with you: let's have additional header with utilities here to help choose device at least (along with other utilities headers in next PRs)

4) I don't see a python in these samples, thus why we need to worry about it in this PR. Let's do not mix concepts

Reply:
I guess this sample might be removed at all, since it shows simple `blur` which is not so interesting.
What we need here are tests. The concern in general about `onevpl` usage it must be as simple as possible, but with possibility to extend. 

If we want to demonstrate how to use specific device with onevpl let's scope separate sample for that...
Like this: https://github.com/opencv/opencv/blob/4.x/samples/va_intel/va_intel_interop.cpp

Reply:
Let me dare to generalize framework concept :) The base concept is provide `api` for user so he/she can focus on algo and stay aside from device/framework/hardware details.

I agree with your point, but probably such details should be shown as separate samples.

Reply:
ok, let's separate samples: according to current situation we need 3 samples:
1) onevpl_specific_device_selection
2) onevpl_single_roi
3) onevpl_bgr

Reply:
Why don't have just `onevpl_specific_device_selection` and keep others simple?

Reply:
it's that I meant:
the one with fined-grained device selection stuff, similarly with current 'single_roi' with DX11 & VAAPI AND with a support matrix for preprocessing VPP;
and two others with "default" selection.
Please keep in mind that `single_roi` now works on DX11 & VAAPI. I would like to keep that also

Reply:
> I disagree, sample must show off flexibility in functionality & configurability and prevent questions like "How I can use my own device in pipeline?" you can put comment like in `onevpl_single_source_roi` about 'default' device selection

I believe we can provide one more super-advanced sample which answers questions like this.
But a dead-simple version is required.
We need to make it easy to use first, otherwise nobody will understand how to get something running quickly.

Reply:
Suggesting a simple solution for now: making OneVPL + OCL sample as simple as possible, so we could reuse the functionality in OMZ. Also renaming `vpl_infer_single_roi` sample to `onevpl_infer_with_advanced_device_selection`